src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncPivotBlockSelector.scala

```diff
@@ -0,0 +1,204 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Cancellable, Props, Scheduler}
+import akka.util.ByteString
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.PeerEventBusActor.PeerEvent.MessageFromPeer
+import io.iohk.ethereum.network.PeerEventBusActor.SubscriptionClassifier.MessageClassifier
+import io.iohk.ethereum.network.PeerEventBusActor.{PeerSelector, Subscribe, Unsubscribe}
+import io.iohk.ethereum.network.p2p.messages.PV62.{BlockHeaders, GetBlockHeaders}
+import io.iohk.ethereum.network.{EtcPeerManagerActor, PeerId}
+import io.iohk.ethereum.utils.Config.SyncConfig
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.FiniteDuration
+
+class FastSyncPivotBlockSelector(
+    val etcPeerManager: ActorRef,
+    val peerEventBus: ActorRef,
+    val syncConfig: SyncConfig,
+    val scheduler: Scheduler
+) extends Actor
+    with ActorLogging
+    with PeerListSupport
+    with BlacklistSupport {
+
+  import FastSyncPivotBlockSelector._
+  import syncConfig._
+
+  val fastSync: ActorRef = context.parent
+
+  def handleCommonMessages: Receive = handlePeerListMessages orElse handleBlacklistMessages
+
+  override def receive: Receive = idle
+
+  def idle: Receive = handleCommonMessages orElse { case ChoosePivotBlock =>
```

Comment:
why not `ChoosePivotBlock` -> `StartPivotBlockElection` ?

Reply:
Because from parent (fast sync) perspective we wnt to choose a pivot block. It doesn't matter if we use an election for that or sth else

Reply:
fair enough, then just a minor one: `ChoosePivotBlock` to `SelectPivotBlock` ? i mean the actor is called `FastSyncPivotBlockSelector`...also, it does matter if is fastSync related? or it could be `PivotBlockSelector` ?

Reply:
Btw, regaring using the name `Start...Election`, take into consideration that programming with actors, behind the scenes, is defining a protocol, so, not sure if not express the way the protocol behaves is best than explicitly mention it. I mean if this was a trait instead of an Actor, i would be agree with you regarding not mention the actual implementation.